internal/ir/xds.go

```diff
@@ -991,6 +993,47 @@ type BasicAuth struct {
 	Users PrivateBytes `json:"users,omitempty" yaml:"users,omitempty"`
 }
 
+// APIKeyAuth defines the schema for the API Key Authentication.
+//
+// +k8s:deepcopy-gen=true
+type APIKeyAuth struct {
+	// Name is a unique name for an APIKeyAuth configuration.
+	// The xds translator only generates one API key filter for each unique name.
+	Name string `json:"name" yaml:"name"`
```

Comment:
Nit: I think we don't need a Name field for APIKeyAuth ir, as only one APIKeyAuth filer in the filter chain.